### DISCUSSION TITLE
Use Rsdoctor in CI/CD

### DISCUSSION THREAD

**Discussion by yifancong** [web-infra-dev/rsdoctor#491] [Show and tell]
In CI/CD, there is often a desire to upload historical reports to the CDN as historical records. Because it's not convenient to achieve instant use in the standard mode, the Brief mode is supported.

## Brief mode

In brief mode, data reports are integrated into a single HTML page, making it easy for users to view historical build data in a summary form within CI/CD and other scenarios.

Enabling Brief Mode[#](https://rsdoctor.dev/guide/start/cicd#enabling-brief-mode)
You can enable Brief mode by configuring the mode.brief option in the Rsdoctor plugin. After the build, Brief mode will generate a report in the build output directory: `[outputDir]/.rsdoctor/report-rsdoctor.html`. You can view the build analysis summary by opening the HTML file in a browser.

In brief mode, no code data is displayed to prevent the page from crashing due to large data sizes.

```ts
rspack.config.js
const { RsdoctorRspackPlugin } = require('@rsdoctor/rspack-plugin');

module.exports = {
  // ...
  plugins: [
    process.env.RSDOCTOR &&
      new RsdoctorRspackPlugin({
        // other options
        mode: 'brief',
      }),
  ].filter(Boolean),
};
```

This feature need @rsdoctor/* > 0.4.0.
More details: https://rsdoctor.dev/guide/start/cicd

## Next todo in CI/CD

When a repository uses the Rsdoctor plugin during the build process, Rsdoctor will support actions that display the report. In future actions, bundle diff support will be added.

**Comment by bpinto** [comment]
I think it would be very useful if there was a github action that 
cached previous build diffs so it could compare the PR build with the base branch build.

It would be able to give a quick summary of the changes and link to a more complete build result otherwise.

**Reply by yifancong** [reply]
> I think it would be very useful if there was a github action that cached previous build diffs so it could compare the PR build with the base branch build.
> 
> It would be able to give a quick summary of the changes and link to a more complete build result otherwise.

Yes, we will start this action in the first quarter of 2025. We are thinking about how to cache the result of the base branch build, maybe need to build twice.

**Reply by bpinto** [reply]
We have a much simpler/limited script compared to rsdoctor that exports a a json file and we cache in using [actions/cache](https://github.com/actions/cache). It needs the build from the base branch and the HEAD branch but it will try to use a cached json file.

Example:

```
- name: Cache result
  id: cache-stats
  uses: actions/cache@v4
  with:
    path: ./asset-sizes.json
    key: webpack-stats_${{ env.STATS_SHA }}
```

and then depending on the cache hit/miss it will execute the rest of the workflow:

```
- name: Get asset sizes
  if: ${{ steps.cache-stats.outputs.cache-hit != 'true' }}
  run: |
    script/fetch-asset-sizes.rb webpack_stats.json > asset-sizes.json
```

I imagine something similar could be achievable here?
